pkg/user/tx_client.go

```diff
@@ -399,20 +426,96 @@ func (client *TxClient) broadcastTx(ctx context.Context, txBytes []byte, signer
 		return nil, broadcastTxErr
 	}
 
-	// save the sequence and signer of the transaction in the local txTracker
-	// before the sequence is incremented
-	client.txTracker[resp.TxResponse.TxHash] = txInfo{
-		sequence:  client.signer.accounts[signer].Sequence(),
-		signer:    signer,
-		timestamp: time.Now(),
+	return resp.TxResponse, nil
+}
+
+// broadcastTx broadcasts the transaction to multiple connections concurrently
+// and returns the response from the first successful broadcast.
+// It uses the primary connection and up to two additional connections.
+func (client *TxClient) broadcastTx(ctx context.Context, txBytes []byte, signer string) (*sdktypes.TxResponse, error) {
+	respCh := make(chan *sdktypes.TxResponse, 1)
+	errCh := make(chan error, len(client.conns))
+
+	g, childCtx := errgroup.WithContext(ctx)
+	var successfulBroadcast sync.Once
+
+	for _, conn := range client.conns {
+		g.Go(func() error {
+			resp, err := client.broadcastSingle(childCtx, conn, txBytes)
+			if childCtx.Err() != nil {
+				if err != nil && !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded) {
+					log.Debug().Err(err).Str("target", conn.Target()).Msg("Broadcast gRPC call failed after context finished")
+				}
+				return nil
+			}
+
+			if err != nil {
+				if !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded) {
+					errCh <- err
+				}
+				return nil
+			}
+
+			successfulBroadcast.Do(func() {
+				select {
+				case respCh <- resp:
+				case <-childCtx.Done():
+				}
+			})
+			return nil
+		})
+	}
+
+	groupErr := g.Wait()
+
+	close(respCh)
+	close(errCh)
+
+	if firstResp, ok := <-respCh; ok {
+		// save the sequence and signer of the transaction in the local txTracker
+		// before the sequence is incremented
+		client.txTracker[firstResp.TxHash] = txInfo{
+			sequence:  client.signer.accounts[signer].Sequence(),
+			signer:    signer,
+			timestamp: time.Now(),
+		}
+
+		// after the transaction has been submitted, we can increment the
+		// sequence of the signer
+		incErr := client.signer.IncrementSequence(signer)
+		if incErr != nil {
+			return nil, fmt.Errorf("increment sequencing: %w", incErr)
+		}
```

Comment:
[nit] incErr seems like an odd name. Can this be renamed to `err`?
[nit] sequence not sequencing

Proposed refactor

```suggestion
		if err := client.signer.IncrementSequence(signer); err != nil {
			return nil, fmt.Errorf("increment sequence: %w", err)
		}
```